mysystehis/main.tex

```diff
@@ -0,0 +1,1155 @@
+ 	 After that, the camera needed to be rotated around with a mouse because i'd like to see what is happening around the character. And the same applied to the character controller script, but this time, i use Input.GetAxisRaw function instead of GetAxis is because i don't want the rotating to be smooth immiately as it can cause a short delay as GetAxisRaw will only return 0,-1 or 1 while GetAxis change gradually from 0 to 1 or 0 to -1.  
+ 	 Moreover, the function Lerp is used here instead of SmoothDamp due to the fact that the smoothdamp function adds the curve which can cause delay, for instance, when i stop moving my mouse, the camera still moving, while Lerp only behaves linear. As now, there's still one more problem, on the mouseY rotation, the camera can be rotated 360 degree which is not i wanted, so i used the function Mathf.Clamp to limit the rotation of the mouseY axis, the minimum, maximum here is -40, 85 degree for the third person camera controller. In the end, to finish my TPS camera controller, i've modified the charactercontroller script a bit by adding cameraT.eulerAngles.y in the targetRotation where cameraT is the MainCamera(the TPS camera), to make sure that the camera will following the player correctly.
+ 	 
+ 	  \begin{itemize}
+	 	\item \bfseries Camera Collision and Occlusion Detection  	 	
+ 	 \end{itemize}
+ 	 In some situations, like in the figure below, when the camera collide with the wall or the wall comes in between the character and the camera, they're called occlusion and collision. When the camera goes through the wall or the ground, it can cause shearing.
+ 	 \begin{figure}[h]
+ 	 	\centering
+ 	 	\begin{minipage}{.4\textwidth}
+ 	 		\centering
+ 	 		\includegraphics[width=1\linewidth]{intructions/camera_cover.png}
+ 	 		\centering
+ 	 		\captionof{figure}{Occlusion example}
+ 	 		\label{fig:test8}
+ 	 	\end{minipage}
+  	\begin{minipage}{.4\textwidth}
+  		\centering
+  		\includegraphics[width=1\linewidth]{intructions/camera_collide.png}
+  		\centering
+  		\captionof{figure}{Collision example}
+  		\label{fig:test9}
+  	\end{minipage}
+ 	 \end{figure}
+  \\[0.15cm]
+  So, the goal here is to handle {\bfseries Collision} and {\bfseries Occlusion} with as little shearing as possible. Let's defined each one of these underlined terms really means. 
+  \begin{figure}[h]
+  	\centering
+  	\begin{minipage}{.4\textwidth}
+  		\centering
+  		\includegraphics[width=1.3\linewidth]{intructions/Occlusion_camera.png}
+  		\centering
+  		\captionof{figure}{Occlusion}
+  		\label{fig:test10}
+  	\end{minipage}
+  \end{figure}
+  \\[0.15cm]
+  Occlusion is when the game is running where the pillar or the wall comes in between the character and the camera view, the character is no where to be seen, and i need to move forward if that happen. Like the figure above, if camera isn't hitting anything but it's view is being obstructed so i have to move forward, the camera can see the player. 
```

Comment:
if the camera is not hitting anything but it's view is being obstructed, I will have to move it forward so that the camera can see the player.